File: src/main/scala/com/magic/core/parser/sql/table/BaseTableSourceParser.scala
package com.magic.core.parser.sql.table

import com.alibaba.druid.sql.ast.statement.SQLTableSource
import com.magic.sqllineageparser.model.{TableNode, TreeNode}

import java.util.concurrent.atomic.AtomicInteger

/**
 * <p>
 *
 * @author Guan Peixiang (guanpeixiang@juzishuke.com)
 * @date 2023/9/12
 */
trait BaseTableSourceParser {
  def process(dbType: String, sequence: AtomicInteger, parent: TreeNode[TableNode], sqlTableSource: SQLTableSource): Unit

}

File: src/main/java/com/magic/core/parser/sql/table/BaseTableSourceParser.java
package com.magic.core.parser.sql.table;

import com.alibaba.druid.sql.ast.statement.SQLTableSource;

import java.util.concurrent.atomic.AtomicInteger;

/**
 * 表源解析器基础接口 (密封接口)
 *
 * @author Guan Peixiang
 * @since 2023/9/12
 */
public sealed interface BaseTableSourceParser
        permits SqlExprTableSourceParser,
        SqlJoinTableSourceParser,
        SqlSubqueryTableSourceParser,
        SqlUnionQueryTableSourceParser,
        SqlWithSubqueryTableSourceParser,
        SqlLateralViewTableSourceParser {

    void process(String dbType, AtomicInteger sequence, SQLTableSource sqlTableSource);
}
